File: q1/rina/q1.cpp
/*

q1
    "I am real sakai lab student"
    の各単語の頭文字をつなげて大文字にして出力するプログラムを書け

    文章を文字列型で定義して始めること

*/

#include <iostream>
#include <string>
#include <cstring>
using namespace std;

int main() {

	int i;
	char str[101];

	for(i=0; i<6; i++) {
		cin >> str;
		if(str[strlen(str)]=='\n') break;
		if(str[0]>='A' && str[0]<='Z') cout << str[0];
		else cout << (char)(str[0]-'a'+'A');
	}

	cout << endl;

	return 0;
}

File: q1/stephanie/q1.py
import doctest

# q1-1
def upper(s):
    lst = s.split()
    for i in range(len(lst)):
        lst[i] = lst[i][0].capitalize()
    return "".join(lst)

s = 'I am a real sakai lab student'
output = upper(s)
print(output) #IAARSLS

# q1-2
def ipv4(s):
    """
    >>> ipv4("192.168.1.1")
    True
    >>> ipv4("333.333.333.333")
    False
    """
    if s.count(".") != 3:
        return False

    lst = s.split(".")
    for i in range(len(lst)):
        if 0 <= int(lst[i]) <= 255:
            return True
    return False

s = input("input an ipv4 address:")
output = ipv4(s)
print(output)

# q1-3
def replace(s):
    return s.replace("sakai", "tetsuya").replace("nakajima", "tatsuo").replace("washizaki", "hironor")

s = input("input (sakai/nakajima/washizaki):")
output = replace(s)
print(output)

doctest.testmod(verbose=1)